interfaces/input_parameters/ring_section.py

```diff
@@ -177,98 +165,134 @@ def __init__(self, section_length, alpha_0,
-        raises a warning.
+        Internal function to check that the input momentum compaction or orbit
+        length is coherent with the synchronous data. If the synchronous data
+        is turn based, the momentum compaction or orbit should have the same
+        length.
+        If the synchronous is time based while the momentum compaction or
+        orbit is turn based, raises a warning.
         '''
 
-        setattr(self, 'alpha_'+str(order), alpha)
+        # attr_name is the attribute to apply to RingSection
+        # attr_name_err is for the warning message
+        if order is not None:
+            attr_name = 'alpha_' + str(order)
+            attr_name_err = attr_name
+        else:
+            attr_name = 'length'
+            attr_name_err = 'orbit_bump'
+
+        setattr(self, attr_name, alpha_or_orbit)
+
+        if (self.synchronous_data.timebase == 'single') and \
+                (alpha_or_orbit.timebase != 'single'):
+
+            warn_message = 'The synchronous data was defined as single element while the ' + \
+                'input ' + attr_name_err + ' was defined turn or time based. ' + \
+                'Only the first element of the program will be taken in ' + \
+                'the Ring object after treatment.'
+            warnings.warn(warn_message)
 
         if (self.synchronous_data.timebase == 'by_turn') and \
-                (alpha.timebase == 'by_turn'):
+                (alpha_or_orbit.timebase == 'by_turn'):
 
-            if (alpha.shape[-1] > 1) and \
-                    (self.synchronous_data.shape[-1] > alpha.shape[-1]):
+            if (alpha_or_orbit.shape[-1] > 1) and \
+                    (self.synchronous_data.shape[-1]
+                     > alpha_or_orbit.shape[-1]):
 
                 raise excpt.InputError(
-                            'The momentum compaction alpha_'+str(order) +
-                            ' was passed as a turn based program but with ' +
-                            'different length than the synchronous data. ' +
-                            'Turn based programs should have the same length.')
+                    'The input ' + attr_name_err +
+                    ' was passed as a turn based program but with ' +
+                    'different length than the synchronous data. ' +
+                    'Turn based programs should have the same length.')
 
         elif (self.synchronous_data.timebase == 'by_time') and \
-                (alpha.timebase == 'by_turn'):
+                (alpha_or_orbit.timebase == 'by_turn'):
 
             warn_message = 'The synchronous data was defined time based while the ' + \
-                'momentum compaction was defined turn base, this may' + \
-                'lead to errors in the Ring object after interpolation'
+                'input ' + attr_name_err + ' was defined turn base, this may' + \
+                'lead to errors in the Ring object after interpolation.'
             warnings.warn(warn_message)
```

Comment:
What if `alpha.timebase == "by_turn"` and `alpha_or_orbit.timebase == "by_time"`?

Reply:
I guess you meant synchronous_data.timebase == "by_turn" and alpha_or_orbit.timebase == "by_time" ?

We could add a comparable warning to the synchronous_data.timebase == "by_time" and alpha_or_orbit.timebase == "by_turn" ?

Reply:
Something like that.  There's at least one combination that isn't currently included anyway.